lib/buffer.js

```diff
@@ -134,41 +141,28 @@ function fromObject(obj) {
     return b;
   }
 
-  if (Array.isArray(obj)) {
-    const length = obj.length;
-    const b = allocate(length);
-    for (let i = 0; i < length; i++)
-      b[i] = obj[i] & 255;
-    return b;
-  }
-
   if (obj == null) {
     throw new TypeError('Must start with number, buffer, array or string');
   }
 
+  if (Array.isArray(obj)) {
+    return fromArrayLike(obj);
+  }
+
   if (obj instanceof ArrayBuffer) {
```

Comment:
If we do this check first, we can skip the `isArray` check and let the `if` block below this, take care of the creation.